core/controllers/classifier.py

```diff
@@ -123,3 +129,61 @@ def post(self):
         classifier_services.mark_training_job_complete(job_id)
 
         return self.render_json({})
+
+
+def update_failed_jobs(job_models):
+    for job_model in job_models:
+        classifier_services.mark_training_job_failed(job_model.id)
+
+
+def fetch_training_data(exp_id, state):
+    exp_model = exp_services.get_exploration_by_id(exp_id)
+    state_model = exp_model.states[state]
+    training_data = state_model.get_training_data()
+    return training_data
+
+
+def fetch_next_job():
+    classifier_job_models = (
+        classifier_services.get_all_classifier_training_jobs())
```

Comment:
I echo this concern. I think querying like this will be much more efficient in terms of retrieval speeds. But, @prasanna08 , current time - last updated time is the total time the job has been PENDING right? So, if this value is less than TTL, why would you mark them failed? Its not crossed its Time to Live yet and is probably still being handled by the VM. Is there something I'm missing here?